pkg/geth/blockchain.go

```diff
@@ -147,6 +147,10 @@ func (blockChain *BlockChain) getPOAHeader(blockNumber int64) (header core.Heade
 	if POAHeader.Number == nil {
 		return header, ErrEmptyHeader
 	}
+	time := POAHeader.Time.ToInt()
```

Comment:
It seems unlikely that the time would be nil when the number is not, but I'm not certain. I think it'd be best to add in a `nil` check just in case?

Reply:
These are `hexutil.Big`s and the `ToInt()` casts won't throw an error on a nil value if I understand correctly (it's just type casting from hexutil.Big to big.Int). 

```go
// ToInt converts b to a big.Int.
func (b *Big) ToInt() *big.Int {
	return (*big.Int)(b)
}
```

Which is why it was originaly okay that the `nil` check is after casting to big.Int

```go
time := POAHeader.Time.ToInt()
if time == nil {
	time = big.NewInt(0)
}
```

I think to avoid confusion it would be best to do the nil check before, though.